hibernate-core/src/main/java/org/hibernate/engine/internal/TwoPhaseLoad.java

```diff
@@ -521,38 +502,9 @@ private static Boolean isEagerFetchProfile(SharedSessionContractImplementor sess
 
 		return null;
 	}
-	
-	private static Boolean isEagerFetchGraph(SharedSessionContractImplementor session, String associationName, Type associationType) {
-		final GraphImplementor<?> context = session.getFetchGraphLoadContext();
-		
-		if ( context != null ) {
-			// 'fetch graph' is in effect, so null should not be returned
-			final AttributeNodeImplementor<Object> attributeNode = context.findAttributeNode( associationName );
-			if ( attributeNode != null ) {
-				if ( associationType.isCollectionType() ) {
-					// to do: deal with Map's key and value
-					session.setFetchGraphLoadContext( null );
-				}
-				else {
-					// set 'fetchGraphContext' to sub-graph so graph is explored further (internal loading)
-					final GraphImplementor<?> subContext = attributeNode.getSubGraphMap().get( associationType.getReturnedClass() );
-					if ( subContext != null ) {
-						session.setFetchGraphLoadContext( subContext );
-					}
-					else {
-						session.setFetchGraphLoadContext( null );
-					}
-				}
-				// explicit 'fetch graph' applies, so fetch eagerly
-				return true;
-			}
-			else {
-				// implicit 'fetch graph' applies, so fetch lazily
-				return false;
-			}
-		}
 
-		return null;
+	private static Boolean isEagerFetchGraph(SharedSessionContractImplementor session) {
+		return session.isEnforcingFetchGraph() ? false : null;
```

Comment:
The above one-liner is the gist of the new implementation. We simply return false. Even if we return true for a Fetch Graph node, it won't make any difference for it will be loaded anyway (the corresponding JOIN has been present in the SQL generated). What really matters is the decision of entity not covered by entity graph.